src/array.c

```diff
@@ -731,11 +731,14 @@ STATIC_INLINE void jl_array_grow_at_end(jl_array_t *a, size_t idx,
     size_t elsz = a->elsize;
     char *data = (char*)a->data;
     int has_gap = n > idx;
-    if (__unlikely((n + inc) > a->maxsize - a->offset)) {
+    size_t reqlen = n + inc;
+    if (__unlikely(reqlen > a->maxsize - a->offset)) {
         size_t nb1 = idx * elsz;
         size_t nbinc = inc * elsz;
-        size_t newlen = a->maxsize == 0 ? (inc < 4 ? 4 : inc) : a->maxsize * 2;
-        while ((n + inc) > newlen - a->offset)
+        // if the requested size is much larger than the current maxsize,
```

Comment:
I think giving the user more control over memory allocation, `resize!()` in particular, would make a lot of sense. Unfortunately, I didn't expect that trivial LAPACK fixes would require that sort of things: the last commit was just a quick check. I'd rather way until the memory allocation problem is addressed by a separate PR and update this one accordingly.